plugins/rating/services/RatingService.php

```diff
@@ -0,0 +1,156 @@
+<?php
+/**
+ * Allows user to manipulate their entry rating
+ *
+ * @service rating
+ * @package plugins.rating
+ * @subpackage api.services
+ */
+
+class RatingService extends KalturaBaseService
+{
+	public function initService($serviceId, $serviceName, $actionName)
+	{
+		parent::initService($serviceId, $serviceName, $actionName);
+		
+		if(!RatingPlugin::isAllowedPartner($this->getPartnerId()))
+		{
+			throw new KalturaAPIException(KalturaErrors::FEATURE_FORBIDDEN, RatingPlugin::PLUGIN_NAME);
+		}
+		
+		if ((!kCurrentContext::$ks_uid || kCurrentContext::$ks_uid == "") && $actionName != "getRatingCounts")
```

Comment:
Better use kSessionBase::isAnonymousSession since you do cover all cases here. 

Use !== for string matching

Reply:
@MosheMaorKaltura  kSessionBase::isAnonymousSession is a class method, not static.

Reply:
$ks = $this->getKs();
!$ks || $ks->isAnonymousSession()

Reply:
@MosheMaorKaltura  slightly different implementation, but should still work.